walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,116 @@
 
 package com.google.sps;
 
+import java.util.List;
+import java.util.Collections;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import com.google.sps.TimeRange;
 
 public final class FindMeetingQuery {
+  private final static int totalMinutes = 1440;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //remove events with no requested attendees
+    HashSet<String> requested = new HashSet<>(request.getAttendees());
```

Comment:
In general, I would recommend to use the interfaces `Set`, `List`. and `Map` instead of their implementations in the variable declaration. It decouples usage from actual implementation. You would have a hard time if you would have to change from `HashSet` to something different.

```suggestion
    Set<String> requested = new HashSet<>(request.getAttendees());
```